lms/djangoapps/ora_staff_grader/tests/test_serializers.py

```diff
@@ -255,3 +263,172 @@ def test_serializer_output(self):
         assert output_data['oraMetadata'] == OpenResponseMetadataSerializer(self.mock_ora_instance).data
         assert output_data['submissions'] == self.mock_submissions_data
         assert output_data['rubricConfig'] == self.mock_rubric_data
+
+
+@ddt.ddt
+class TestScoreFieldAndSerializer(TestCase):
+    """ Tests for ScoreField and ScoreSerializer """
+
+    def test_field_no_values(self):
+        """ An empty dict passed to the field should return None """
+        assert ScoreField().to_representation({}) is None
+
+    @ddt.data('pointsEarned', 'pointsPossible')
+    def test_field_missing(self, missing_field):
+        """ One missing field should result in the field defaulting to 0 """
+        value = {'pointsEarned': 30, 'pointsPossible': 50}
+        del value[missing_field]
+
+        expectedValue = dict(value)
+        expectedValue[missing_field] = 0
+
+        actualValue = ScoreField().to_representation(value)
+        assert expectedValue == actualValue
+
+    def test_field(self):
+        """ Base serialization behavior for ScoreField """
+        data = {
+            'pointsEarned': 20,
+            'pointsPossible': 40
+        }
+        representation = ScoreField().to_representation(data)
+        assert representation == data
+
+    def test_serializer_no_values(self):
+        """ Passing the ScoreSerializer an empty dict should result in two zeroes for the field """
+        data = ScoreSerializer({}).data
+        assert data == {'pointsEarned': 0, 'pointsPossible': 0}
```

Comment:
This is a change in contract. I don't dislike the change but it also isn't what we had designed initially. Initially empty score should (intentionally) return `None` to let the frontend know there's no score data.

Reply:
Yes. https://github.com/edx/edx-platform/pull/29268#discussion_r743882795

The ScoreField returns None when given an empty dict, but the Serializer apparently is not allowed to. This is just a test for completeness, not actual business-case usage, and this scenario should never actually exist, but I needed to choose /something/ for this scenario. I could make this return an empty dict if you think this is too confusing

Reply:
There's a benefit of returning `None`/`{}`: we make it abundantly clear to the frontend that we don't have grade data, instead of passing default values. That said, this is probably fine, and definitely a nit/edge-case, running on the assumption that the frontend uses `gradingStatus` instead of `score` presence/absence to determine if something has a score.

Reply:
That makes sense I'll make this return an empty dict so that even though this should never see the light of day it conceptually aligns with the API